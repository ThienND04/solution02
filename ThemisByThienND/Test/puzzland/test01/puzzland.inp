32
14 75
L O T V U W M R Y Z I C N J 
R I
O Z
T N
Z J
T U
L U
W N
Y J
T W
W R
V R
L J
M I
M C
V W
V J
R Y
O J
V Z
N J
Y I
O T
U R
Z N
V Y
O I
O N
T V
I J
M J
W I
T M
O U
T J
V I
O W
W J
I C
T Z
T I
L Z
U N
L M
U I
L T
L W
L V
Y N
O Y
M R
Y Z
R C
U M
L I
U C
L C
W C
O C
W M
U J
Y C
R Z
L Y
L O
L R
U Y
C N
I N
U W
O M
T Y
V C
T R
V U
C J
13 51
K B W L O Y J Q A M T U Z 
M Z
Q T
O Y
O T
L Y
Y A
W T
W L
L M
T Z
L Z
A T
L T
W J
K T
A M
L Q
B U
K L
K B
J Z
B L
B A
K Y
J M
O Z
A Z
L O
B Z
K O
Y J
M U
Y T
Y M
Y U
U Z
O J
W Q
K Z
B Y
B O
B Q
L U
K A
W O
B M
Q M
W M
Q Z
W Z
K J
15 86
Q J A E B R K V U S Y G P Z W 
A R
J Y
R G
Q W
V Z
A G
V G
E G
V Y
A K
K P
B K
S P
V S
B V
Q Y
B P
A W
U P
J A
Y P
B Y
U Y
R Y
Y Z
G Z
J S
Q G
B S
B W
J P
G P
V U
A S
K Y
B U
E B
E R
Q A
K U
R U
J Z
E W
Q R
R S
E P
B Z
Z W
A P
S G
U W
S Y
A Z
E U
E Y
S Z
A E
B G
K S
Q P
U Z
P W
E S
R Z
Q U
K V
J R
Q J
A Y
U G
G W
P Z
K W
Y G
Q E
R P
A U
A B
Y W
B R
K Z
J K
J G
U S
J U
R V
12 50
F N D Q I H C G M Z Y R 
Q I
C Y
C R
Z R
H Y
G Z
D G
C G
I R
F Q
F D
H C
F G
F C
F R
F Y
D M
I H
F M
N Z
N R
I Y
M Z
D I
G Y
N Q
D Z
N G
D R
D H
H M
Z Y
F H
I G
M R
D Q
D C
F Z
I C
N C
F N
N I
Q Z
H G
N M
Q H
Q R
G M
H R
Q G
15 75
X U F Y E G M J O B V Z S P T 
G O
Y O
F E
X M
U E
X Z
Y J
Y G
F B
Z P
X U
M P
X J
M S
U Z
P T
Y S
J S
F Z
J P
F Y
S T
J Z
J V
V P
X S
B S
V T
U S
G S
F M
E B
B Z
E J
U T
X E
U M
U B
X Y
M T
F J
J T
M Z
U G
F S
E T
E G
X G
V Z
G V
F G
G M
O V
U O
U F
O B
F P
O P
X V
E P
X F
M O
G T
E V
Y B
S P
F O
O T
U V
G P
J O
O S
G J
Z S
G Z
13 77
Q B R G T H N L A S W J C 
H W
S C
N J
N L
L A
T C
G H
R W
B L
J C
G A
Q W
Q R
T A
T N
Q G
N S
S W
A W
R A
B H
L J
B A
L W
H A
G C
N A
G J
T H
A J
T S
H N
R J
G S
B S
A S
Q S
B N
R S
Q J
Q B
L C
B C
T W
T L
A C
B R
Q A
S J
R G
R N
H S
B T
N W
R H
Q N
B G
B W
G L
Q H
G W
W J
Q T
Q C
L S
H L
R C
H C
W C
G T
N C
Q L
R T
H J
R L
T J
G N
15 60
U D L N B K W S H I A V Y E R 
B W
A R
B H
N W
K E
B R
I V
W S
S V
D K
S R
U Y
A V
D Y
H I
A Y
U I
N A
U K
K S
U S
B I
W Y
B A
U N
I A
N K
I Y
U V
K V
K W
W R
D A
I E
A E
D L
V R
L B
L Y
L A
B S
D N
U E
N I
D V
L K
V Y
L W
E R
D B
B Y
B E
L N
S Y
W H
W E
K H
U D
D R
U R
13 60
B D T U L M Z C O F G H P 
Z H
L H
B L
O P
L C
D U
D M
L Z
B U
B G
M P
U L
B F
C G
D T
D C
O F
M G
D F
B D
L O
H P
L G
U H
F H
U G
M Z
T L
M O
D P
M F
B T
T P
T M
T F
M C
M H
B P
B H
D O
B Z
T C
D L
F P
U P
T H
D H
U C
O G
G H
D Z
L F
U Z
L M
C O
G P
B O
Z O
U O
F G
14 84
A Y M V J K C O B X D T F P 
Y T
O X
C D
V P
V B
B P
V C
M V
J B
B T
A X
X P
V F
A D
V K
Y P
A T
Y J
O B
B F
C X
O T
M F
C O
V J
Y M
M K
M O
C T
M T
M J
A O
V X
J K
K F
Y V
X F
F P
J C
M X
J X
M C
T F
K X
O D
Y D
Y C
V O
M P
Y K
X T
D P
Y X
X D
A J
J D
T P
A P
K O
J P
B D
K T
K C
A F
D T
O P
J T
J F
K D
A K
C B
A M
V D
M D
Y B
O F
A Y
C F
M B
A C
K P
A B
Y O
V T
12 58
G R O B H F T V Z U I P 
R Z
G R
G Z
R P
B Z
F T
V Z
G B
G F
T V
T Z
H T
O P
G I
B U
I P
V P
R O
O V
O U
H F
H P
Z P
F U
H Z
R F
R V
O H
Z I
V I
B V
G V
G P
U P
T P
Z U
H I
B T
O B
F V
H V
O I
T I
R I
R T
T U
F I
G O
R U
U I
R H
O T
R B
B P
B F
O Z
G T
F Z
12 65
X B C O Z H D Q M F T I 
Q M
Z T
D Q
H M
F I
H D
X O
D T
C F
Z Q
C D
B C
F T
X F
B T
Q I
O H
X H
B Q
B O
X D
H I
M T
D F
D M
X T
B D
O M
B I
X I
B H
T I
Z H
D I
X Z
C Q
Z D
X Q
C T
B M
H T
C H
O I
C M
X B
O T
Q F
H Q
Q T
O Z
C O
C I
H F
O F
O D
M F
X M
B F
B Z
M I
O Q
C Z
Z M
Z I
X C
13 45
S L H J R F Z P D B A N I 
S A
R Z
L F
Z A
Z N
J A
R D
H I
S B
R A
N I
L D
A I
F Z
R I
L P
L A
R P
S J
H J
J D
A N
S F
S D
L I
L R
D A
L H
L Z
F N
J N
H P
F I
S P
B N
F D
J F
L N
P A
J Z
Z B
P N
S I
F A
Z I
15 59
T S C L K B F Y D H Q P I V E 
T V
L P
F D
T B
L Q
T P
S B
B Q
S D
L H
C K
B D
C Y
T H
K B
I E
F E
Q V
F Q
K D
C E
T K
T I
C V
S H
D H
L E
T S
Y I
B H
I V
Q E
S Q
T F
H Q
K E
Y V
K I
B E
F P
H V
B Y
T Y
C H
C P
S I
Y D
L F
B P
S V
S K
K P
D P
S L
K Y
H E
C D
L V
S Y
13 40
D B G X W K L H E N Q Y Z 
H Y
K L
Q Y
D E
G Z
G Y
K N
L Q
K H
X K
B Z
W Q
E N
K Z
W E
H Q
H E
D L
B L
G E
Q Z
L H
G W
B Q
B G
H N
B X
D Z
W Y
B N
X H
G H
B W
X E
W K
L E
E Y
N Z
L Y
G X
15 81
P R Z T L Q G O J E V W S C A 
G J
O E
T L
W S
T S
P A
Z A
Z V
P W
R E
E S
T W
R S
Z W
O W
Q C
Q O
O C
P C
Q E
V W
R A
T J
R C
P T
V S
E A
J A
Z S
Z L
P V
Z C
R Z
E V
V A
L Q
R Q
Q J
P Q
R W
O J
Q G
R T
W A
Z G
L O
L S
P O
V C
W C
J W
R J
G V
L E
O S
Z Q
L J
Q S
T C
O V
T V
R V
T E
J E
L C
Q W
P R
G A
Q A
P S
R O
L W
P J
S C
R L
T A
G C
T O
P Z
P E
E C
15 67
P X W K T R A D Z L I E S Y H 
A E
T R
R I
I S
K T
W K
R Z
W I
S H
A H
R Y
T Y
R E
I E
A Z
K D
K R
K S
P W
I H
W A
W S
K A
L S
X Z
T S
P T
R D
X I
X E
P A
D I
R H
P H
L I
E Y
Z E
W L
X K
K L
X H
Z H
W D
X A
D L
K E
K I
X D
A L
A S
P E
L Y
D Z
Z I
X L
X Y
A D
T L
S Y
K H
E S
L E
P I
P R
T D
P L
R L
12 40
Y J D V N B F Z T M Q W 
V B
F T
D N
B Z
J B
F Q
D W
J N
Y F
B Q
V W
T Q
Y Q
B M
Z W
D F
Z T
J W
B T
Y V
F Z
J Q
J V
N T
N Z
N B
V Q
V M
Q W
M W
J Z
Y B
F W
B F
V F
Z Q
T W
Y M
M Q
N M
13 70
R G I X Z L F K W A U O P 
R F
R P
L O
R Z
L K
X Z
X W
F A
Z W
Z U
X O
I U
R L
R A
O P
K A
F P
R O
F K
G F
R W
X U
F U
W O
X K
W A
U O
G X
G U
Z O
R I
Z K
I X
I L
I Z
I F
L A
I P
K W
L W
G I
F W
R U
I A
I W
R K
K U
Z A
L F
K P
G A
Z L
G W
X F
I O
Z P
U P
X A
G Z
F O
R G
W U
A U
L U
G P
K O
G O
X L
G K
R X
14 91
I A O N P G H D B T W M S F 
A B
A T
G T
P T
O B
G D
B W
D S
I W
O D
D B
H T
G B
G S
H F
P H
H M
P B
N G
A O
N W
I P
N M
I F
N S
B T
P W
O S
N F
O G
I T
G F
W S
N B
A M
I O
A D
H W
A P
W F
M S
T S
H S
N H
P D
A F
D W
O F
T M
A G
O W
N T
A N
A H
P M
P S
O M
I S
D T
O H
D F
O T
P G
O P
H B
G M
B F
A S
I G
G W
I D
I H
I M
S F
N P
P F
H D
T W
I B
M F
B M
I N
G H
I A
D M
W M
N D
T F
O N
B S
A W
14 56
U S W E B C T N O A R X J I 
U B
A J
E T
B I
W C
C A
C R
T R
N J
S C
S N
T J
C X
E O
E X
S A
B R
E C
E J
O I
B N
W I
X I
S W
U I
N I
U C
O R
U W
N O
C T
O J
A I
B J
R I
O X
E I
U N
U E
E B
J I
B C
U T
B A
S J
X J
T I
S E
S R
A X
W X
T O
N X
E N
S I
S X
12 52
S Y C L F G I A H W T Q 
Y Q
L H
G Q
C F
L A
S G
I W
Y I
C L
C Q
H W
A W
C W
Y F
I Q
S C
G W
L T
S Y
Y A
C I
S H
H T
Y T
H Q
F I
I T
G A
S Q
Y W
F T
A Q
A H
F Q
W Q
F H
Y C
L Q
S A
Y L
C T
S L
F A
S I
A T
L I
S W
F W
S T
G I
C G
Y G
14 59
I N H F R W X C E B V J P T 
V T
N C
B T
F E
I E
J T
F W
N F
H X
X T
I R
W B
C T
H T
N H
N J
I F
V P
F V
I B
H J
C B
H R
F X
X J
C E
R P
I V
P T
H V
F P
X E
F R
X P
I C
J P
H C
E J
I P
B V
R W
R V
B J
F B
N R
I X
H W
I T
X V
E P
W J
R X
N X
R B
H F
C J
R E
W E
I W
14 87
S B P Q D G T E L U I O A H 
D E
I A
D T
G I
O H
Q H
G H
B G
G E
P H
E A
B H
Q L
L O
B I
D U
B A
I H
P Q
E H
Q E
D H
B E
Q G
Q D
P D
U I
P I
P O
T A
E I
D L
S U
G A
S B
Q O
B T
E U
G U
E L
L A
I O
U H
L H
S I
B L
Q T
P L
T U
P U
Q I
T H
S A
G O
S Q
Q U
B P
S D
B U
S L
S T
T L
P E
Q A
P G
S O
L I
O A
D O
S H
L U
D I
P T
E O
D A
B D
U O
S G
A H
S P
B Q
S E
U A
G T
G L
T O
T E
15 82
J T I Q F B U V O E L D X G A 
E A
J T
J F
B X
J Q
D X
B O
V G
X G
B U
G A
E G
Q V
T U
J O
T G
D A
I O
U E
J I
V D
Q G
O E
U X
F U
I U
T D
I G
V X
V L
T I
F L
E L
V O
O A
B G
Q A
I F
O L
T Q
Q F
E X
B A
Q O
J B
F E
O D
U G
Q X
T E
U A
J V
I B
F A
U V
J U
B V
U D
Q L
U O
V A
L A
F B
I X
Q B
Q E
J E
F X
O G
F G
O X
B D
F O
I A
J A
L X
X A
T L
I V
J G
L G
T X
14 72
B I V X S W G P N M T Z O C 
X O
P N
T C
S N
B G
G M
G Z
X W
N T
O C
X P
T O
S T
G C
I G
I V
G O
V S
X M
W M
P M
N C
I C
G N
B P
X N
V Z
S Z
X S
S W
S M
T Z
S C
I O
N O
P O
X G
W Z
P C
W G
S P
B T
B W
I P
M T
B N
V O
B V
I S
X Z
W P
M O
M C
N Z
N M
V T
G T
B M
I M
V P
W N
I N
I W
V C
V W
B Z
I T
W T
Z C
Z O
G P
B O
13 69
D I P Q F T L G Y S E Z O 
F G
T S
Q O
T Y
T Z
F Y
P G
S O
I Y
P Z
I F
P L
D Z
Y S
D E
F O
D G
D O
Q L
L Y
D I
T E
F E
D Q
S E
E O
G Z
T L
P F
D Y
I T
Y O
G Y
I O
D S
D L
I L
Q E
I G
I S
L S
F L
P Y
Q F
F T
E Z
G O
L O
Z O
D T
I Z
Q Z
S Z
Q G
G E
Q T
Y Z
P Q
L G
G S
F S
T O
P O
T G
L Z
Y E
P T
F Z
D P
13 62
S B Q I W H G J K L N X F 
S N
L F
G X
H G
Q N
B H
S K
I H
K N
B F
W X
L N
H N
J K
G K
W J
J N
I X
Q X
B G
N F
H F
S F
K F
W F
B Q
Q I
B W
S G
H K
Q G
I J
S I
S B
B X
H L
I N
Q L
W G
N X
Q H
S Q
J F
I G
S W
S J
L X
H J
B N
H X
I F
Q J
J L
W N
W L
I W
G L
W K
Q F
S L
I L
B I
15 76
B I L F Z K G R O C X P M Q A 
O Q
Z R
Z C
Q A
L K
R O
K C
O C
L Q
R M
F Q
C P
K R
Z A
B K
L G
R C
O X
F X
C Q
F M
M A
B F
F O
B G
X A
B A
L Z
C X
X Q
G A
Z M
K M
P M
B O
R A
L M
L A
I G
I K
B P
P A
G Q
B M
C M
O A
F K
B Z
F G
L R
B Q
X P
B X
I L
I F
C A
I X
G R
K P
G O
Z P
L F
F P
I R
L O
I Z
I O
B C
Z K
K X
M Q
Z G
B I
B L
F A
Z X
13 61
Q D H K J G R W I B Z V Y 
G V
D K
R B
K J
D J
D Z
R W
G I
J G
I B
R Z
H J
H W
Q Y
H Y
K G
J Y
K Y
Q V
G Y
Q D
W I
Q W
D H
G R
G Z
Q R
H I
K R
I Z
H R
R Y
J W
Q I
W B
J Z
H V
Q J
H G
B Y
K I
D I
G W
Q G
Z V
W Z
J I
K B
K W
Q Z
H B
R I
D R
G B
K Z
B V
I V
D W
Z Y
W Y
Q H
14 58
W U G F V T L K M D Z O H C 
Z O
G D
Z H
L D
U C
K M
M Z
D Z
G L
M O
T Z
O H
U G
T H
G V
U V
W K
M H
T O
T M
Z C
F T
G M
U O
K O
K C
H C
V Z
W O
U F
W M
K Z
G O
V H
M D
L M
F V
W H
U Z
G K
L C
G H
G C
V K
W T
T D
W L
W C
L K
D H
F L
U M
F D
V M
F C
G T
U D
F O
13 63
N E I B Z X T F Q P J K W 
Z P
Q J
N J
P J
T P
N W
Z K
I Q
N I
P K
E I
E Z
B Z
E F
E B
T Q
I F
N P
X K
F K
B Q
B K
F Q
T F
N Q
N B
B W
B F
B J
Z Q
F P
J W
N F
F J
T W
P W
Q W
T K
E X
E W
J K
N Z
X Q
Z J
E P
I K
B P
N T
I J
I X
X P
Q K
X W
E T
I T
N E
I P
N X
I B
Z W
X F
I Z
K W
13 65
W K V N I Z F J S O X T L 
K F
K S
I T
S X
I Z
K V
W Z
V Z
I X
Z J
W L
K J
N X
J O
F L
W V
V J
J X
Z F
V X
N L
O X
N Z
X L
V N
F T
J T
W I
W X
Z O
F J
W O
K T
I S
Z T
I L
N T
Z L
K I
K L
J L
O L
I O
V O
N I
T L
W S
K O
X T
V S
N F
K X
W N
W T
V I
F X
S L
F O
N O
I F
N J
W F
J S
W K
I J
