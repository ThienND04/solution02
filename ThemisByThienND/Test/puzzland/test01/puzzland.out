LCIMONTRUWVYZJ
KABLMJOWQTYUZ
QABEGJKPRSUVYZW
FCDGHIYZMNQR
XEBFGJOYSMPVZUT
QABGHJLNRSTWC
UDABEIHKLNWSYVR
BDCGFHLOUZMTP
ABCDJFKMOTVXYP
GBFHIORTUZVP
XBCDFHMOQTZI
SADFJHLPNBZRI
TBDCHLFPKISYVQE
DEGBLQWYHXKNZ
PCEJGQLORSTVWZA
PADIEKLWSTRYXZH
YBFDNJVMQTZW
RAFGILKOUWXZP
IABDGHMNOPSTWF
UBACEJNOTRSWXI
SAFCGILHTYWQ
IBCEFHJWRNXPVT
SABDEGILOPQTUH
JBDOEFGITLQVUXA
BGIMNOPSTVWXZC
DEFGILPQTSYZO
SBGHIJKNLQXWF
BCKFGILMPXQORZA
QDHBGIVZJKRWY
WHDFLGKMOTZVUC
NBEFIJKTPXQZW
WFIJKONTXSVZL
